packages/spread/operation/framework.py

```diff
@@ -23,6 +23,7 @@ def __init__(self, graph: GraphDatabase.driver, engagement_handle: str, operatio
         self.engagement_handle = engagement_handle
         self.operation_name = operation_name
         self.logging_identifier = f"{self.operation_name}[{self.engagement_handle}]"
+        # TODO: Add `self.engagement_data = None` here. Used in all sub-classes.
```

Comment:
Good idea, let's implement this at the framework level and remove the line from the sub-operations